packages/@aws-cdk/aws-ecr/lib/repository.ts

```diff
@@ -307,10 +209,79 @@ export interface RepositoryProps {
   readonly retain?: boolean;
 }
 
+export interface RepositoryAttributes {
+  readonly repositoryName: string;
+  readonly repositoryArn: string;
+}
+
 /**
  * Define an ECR repository
  */
 export class Repository extends RepositoryBase {
+  /**
+   * Import a repository
+   */
+  public static fromRepositoryAttributes(scope: Construct, id: string, attrs: RepositoryAttributes): IRepository {
+    class Import extends RepositoryBase {
+      public readonly repositoryName = attrs.repositoryName;
+      public readonly repositoryArn = attrs.repositoryArn;
+
+      public addToResourcePolicy(_statement: iam.PolicyStatement) {
+        // dropped
```

Comment:
Should we make something similar to `Grant.drop()` here?

I guess `Grant` is not the right location but it would be good if it were a standardized piece of code.